sdk/core/azure-core/azure/core/credentials.py

```diff
@@ -28,3 +29,45 @@ def get_token(self, *scopes, **kwargs):
     from collections import namedtuple
 
     AccessToken = namedtuple("AccessToken", ["token", "expires_on"])
+
+__all__ = ["AzureKeyCredential"]
+
+
+class AzureKeyCredential(object):
+    """Credential type used for authenticating to an Azure service.
+    It provides the ability to update the key without creating a new client.
```

Comment:
What is the scenario that user needs to update key after creation?
And do we want to support per-request key? (get key from context in on_request method)

Reply:
we want to support key rolling / rotating, which is why we have the ability to update keys.

I don't think there're any plans for adding a per-request key, with this pr i'm just trying to move the key credential we already had in place for search and text analytics into azure core for shared usage